wolfcrypt/src/asn.c

```diff
@@ -23482,13 +23570,40 @@ static int EncodeExtensions(Cert* cert, byte* output, word32 maxSz,
                     cert->extCustom.oid, cert->extCustom.oidSz);
             SetASN_Buffer(&dataASN[CERTEXTSASN_IDX_CUSTOM_STR],
                     cert->extCustom.val, cert->extCustom.valSz);
-        }
-        else {
+        } else
+    #endif
+        {
             /* Don't write out custom OID. */
             SetASNItem_NoOut(dataASN, CERTEXTSASN_IDX_CUSTOM_SEQ,
                     CERTEXTSASN_IDX_CUSTOM_STR);
         }
+
+        i = 0;
+    #ifdef WOLFSSL_CUSTOM_OID
+        for (; i < cert->customCertExtCount; i++) {
+             int idx = CERTEXTSASN_IDX_START_CUSTOM + (i * 4);
+             word32 encodedOidSz = MAX_OID_SZ;
+             idx ++; /* Skip one for for SEQ. */
+             /* EncodePolicyOID() will never return error since we parsed this
+              * OID when it was set. */
+             EncodePolicyOID(&encodedOids[i * MAX_OID_SZ], &encodedOidSz,
+                             cert->customCertExt[i].oid, NULL);
+             SetASN_Buffer(&dataASN[idx], &encodedOids[i * MAX_OID_SZ],
+                           encodedOidSz);
+             idx++;
+             SetASN_Boolean(&dataASN[idx], cert->customCertExt[i].crit);
```

Comment:
Criticality is not present in encoding if boolean is false.

Reply:
RFC 5280 states " An extension includes the boolean critical, with a default value of FALSE."  I am not aware of any convention in ASN.1 that says a value must not be present if it is the default value.

That said, it is easy enough to do `dataASN[idx].noOut = 1;`

Reply:
DER encodings of SEQUENCE and SET components shall not include an encoding for any component value which is equal to its default value.
Section 11.5
https://www.itu.int/ITU-T/studygroups/com17/languages/X.690-0207.pdf

Reply:
Wow!!! Thanks for the reference!!